src/raw/mod.rs

```diff
@@ -3065,6 +3306,125 @@ impl<T> Iterator for RawIter<T> {
 impl<T> ExactSizeIterator for RawIter<T> {}
 impl<T> FusedIterator for RawIter<T> {}
 
+/// Iterator which returns an index of every full bucket in the table.
+///
+/// For maximum flexibility this iterator is not bound by a lifetime, but you
+/// must observe several rules when using it:
+/// - You must not free the hash table while iterating (including via growing/shrinking).
+/// - It is fine to erase a bucket that has been yielded by the iterator.
+/// - Erasing a bucket that has not yet been yielded by the iterator may still
+///   result in the iterator yielding index of that bucket.
+/// - It is unspecified whether an element inserted after the iterator was
+///   created will be yielded by that iterator.
+/// - The order in which the iterator yields indices of the buckets is unspecified
+///   and may change in the future.
+pub(crate) struct FullBucketsIndices {
```

Comment:
Could `RawIter` be built on top of this to avoid code duplication? Assuming that doesn't impact compilation times too much due to the additional layer of inlining needed.

Reply:
Actually I think this would improve compile times since the code for iteration would only need to be instantiated once.

Reply:
I was thinking that `FullBucketsIndices` could be changed to take a size field and thus maintain a pointer to the bucket. There could be some performance impact due to the use of indices that `RawIter` could be sensitive too.

I'd try these changes on top of this PR though.

Reply:
I will try 😄

Reply:
Oh, that `reflect_toggle_full` method of `RawIter`.  It causes the `FullBucketsIndices` size to bloat.  Okay, I'll go to bed, maybe tomorrow I'll come up with something adequate 🤔